### PR TITLE
Removing warnings from libencrypt and libdecrypt

### PR DESCRIPTION
**Pull request by behzadmehmood-rs**
#### Description

This change simply removes warnings due to type casting.

#### Motivation and Context

It is necessary for a warning free CI.

#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by behzadmehmood-rs** [comment]
@tangxifan some QoR tests are failing in CI which are not related to the changes in this PR.

**Comment by tangxifan** [comment]
@behzadmehmood-rs Thanks. The CI failures are expected then. Good job.

**Comment by tangxifan** [comment]
@Tulong4Dev Please help on reviewing this PR.

File: libs/libencrypt/src/encryption.cpp [hunk 1]
```diff
@@ -116,47 +181,52 @@ std::string Encryption::base64_encode(const std::string& input) {
...
+        EVP_PKEY_free(publicKey);
         return false;
     }
 
-    // Base64 encode session key
-    std::string base64EncryptedSessionKey = base64_encode(encryptedSessionKey);
+    unsigned char iv[EVP_MAX_IV_LENGTH];
+    if (RAND_bytes(iv, sizeof(iv)) != 1) {
+        std::cerr << "Error generating IV." << std::endl;
+        return -1;
```
Comment (Tulong4Dev) [inline libs/libencrypt/src/encryption.cpp:217]: should return a Boolean  here.

File: libs/libencrypt/src/encryption.cpp [hunk 2]
```diff
@@ -116,47 +181,52 @@ std::string Encryption::base64_encode(const std::string& input) {
  */
 bool Encryption::encryptFile(const std::string& publicKeyFile, std::string& filePath ) {
 
-    // Load public key
-    RSA* publicKey = loadPublicKey(publicKeyFile);
+    OpenSSL_add_all_algorithms();
+    ERR_load_crypto_strings();
+    unsigned char sessionKey[16];
```
Comment (Tulong4Dev) [inline libs/libencrypt/src/encryption.cpp:186]: can we avoid using constant number here ("16") directly?  we can use a meaningful constant variable to represent it.

**Review by Tulong4Dev — CHANGES_REQUESTED** [review]
did a quick screening and suggested to make some minor fix.
 meanwhile, i'm afraid there could be some memory leakage which i suggest to use valgrind to detect and verify it, if possible.

**Comment by behzadmehmood** [comment]
> did a quick screening and suggested to make some minor fix. meanwhile, i'm afraid there could be some memory leakage which i suggest to use valgrind to detect and verify it, if possible.

I have made the changes, now I will run it with valgrind.

**Comment by behzadmehmood** [comment]
> > did a quick screening and suggested to make some minor fix. meanwhile, i'm afraid there could be some memory leakage which i suggest to use valgrind to detect and verify it, if possible.
> 
> I have made the changes, now I will run it with valgrind.

@Tulong4Dev , I have also updated the code to avoid memory leaks.

**Review by Tulong4Dev — COMMENTED** [review]
LGTM. In addition, I'd suggest to add a test with encrypt/decrypt so VTR's Valgrind test flow could help to detect potential memory leak.

**Comment by behzadmehmood** [comment]
> LGTM. In addition, I'd suggest to add a test with encrypt/decrypt so VTR's Valgrind test flow could help to detect potential memory leak.

Is it okay to put the test data and code in `libdecrypt`?

**Comment by behzadmehmood** [comment]
> LGTM. In addition, I'd suggest to add a test with encrypt/decrypt so VTR's Valgrind test flow could help to detect potential memory leak.

@Tulong4Dev I have added a test but I could not update workflow file properly. Can you please suggest how should I run the test?

File: libs/libdecrypt/test/src/test.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#include "test.h"
+
+#ifndef PUBLIC_KEY_FILE
+#define PUBLIC_KEY_FILE "public_key.pem" // Default public key file
+#endif
+
+// Function to create a sample XML file for testing
+void createTestXMLFile(const std::string& filePath) {
```
Comment (tangxifan) [inline libs/libdecrypt/test/src/test.cpp:8]: @Tulong4Dev Please check if the test provided is sufficient. If a local run on the unit test is o.k., I suggest to merge this PR. This PR is currently a blocking factor on another PR #2637

**Review by tangxifan — APPROVED** [review]

**Review by Tulong4Dev — APPROVED** [review]

**Pull request merged**